### PR TITLE
fix: lexsort_to_indices should not fallback to non-lexical sort if the datatype is not supported

### LINKED ISSUE #6226: lexsort_to_indices should not fallback to non-lexical sort if the datatype is not supported
**Describe the bug**

Comet encountered some errors on Spark test like following. It is because DataFusion SortExec calls `lexsort_to_indices` and the function fallbacks to `sort_to_indices` for single column case. However, `sort_to_indices` doesn't support all data types, e.g., struct. For unsupported data types, `lexsort_to_indices` should not fallback to `sort_to_indices`.

```
[info] - vectorized reader: struct *** FAILED *** (515 milliseconds)        
[info]   org.apache.spark.SparkException: Job aborted due to stage failure: Task 0 in stage 644.0 failed 1 times, most recent failure: Lost task 0.0 in stage 644.0 (TID 862) (192.168.86.44 executor driver): org.apache.comet.CometNa
tiveException: Compute error: Sort not supported for data type Struct([Field { name: "_1", data_type: Int32, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }, Field { name: "_2", data_type: Utf8, nullable: true, dict_id: 0, dict_is_ordered: false, metadata: {} }])                                                                                                                                                                                    
```

**To Reproduce**

**Expected behavior**

**Additional context**

### PR DESCRIPTION
**Pull request by viirya**
# Which issue does this PR close?

Closes #6226.

# Rationale for this change
 

# What changes are included in this PR?

# Are there any user-facing changes?

### CODE REVIEW

File: arrow-ord/src/sort.rs [hunk 1]
```diff
@@ -4203,4 +4237,43 @@ mod tests {
...
+                float.clone() as ArrayRef,
+            ),
+            (
+                Arc::new(Field::new("c", DataType::Int32, false)),
+                int.clone() as ArrayRef,
+            ),
+        ]);
+
+        assert!(!can_sort_to_indices(struct_array.data_type()));
+        assert!(sort_to_indices(&struct_array, None, None).is_err());
```
Comment (crepererum) [inline arrow-ord/src/sort.rs:4258]: might be helpful to check the error, either via `matches!`/`assert_matches!` or via `err.to_string()` so we make sure that the right error is returned on not some abstract "internal error" or something.

Comment (viirya) [reply arrow-ord/src/sort.rs]: Makes sense. Checked the error in the test.

**Review by crepererum — APPROVED** [review]

**Comment by alamb** [comment]
Thanks @viirya  and @crepererum

**Comment by viirya** [comment]
Thank you @crepererum @alamb

**Pull request merged**